File: src/main/resources/public/js/shared/service/SectorService.js
'use strict';

angular.module('virtualcloset').service('SectorService', SectorService);

SectorService.$inject = [ 'BaseService' ];

function SectorService(BaseService) {
	var self = this;
	self.api = 'sector';
	
	self.get = (fn) => BaseService.get(self.api, fn);
}

File: src/main/java/vidias/virtualcloset/service/SectorService.java
package vidias.virtualcloset.service;

import java.util.Collection;
import java.util.LinkedList;
import java.util.List;

import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.stereotype.Service;

import vidias.virtualcloset.model.Sector;
import vidias.virtualcloset.repository.SectorRepository;

@Service
public class SectorService {
    
    @Autowired
    private SectorRepository sectorRepository;
    
    public Collection<Sector> getAll() {
        List<Sector> sectors = new LinkedList<>();
        sectorRepository.findAll().forEach(sectors::add);
        return sectors;    
    }
    
    public Collection<Sector> getAllNonOptional() {
        return sectorRepository.findAllNonOptional();
    }
}
